Lauren jenkins

laurendiperna opened the pull request:
This pull request corresponds to jira tickets PUBDEV-3167, PUBDEV-3121, and PUBDEV-3076 which asked for variable importances (for gbm, drf, and dl) and standardized coefficient magnitudes (for glm) plots for the R and Python API (which includes pyunits/runits for each). There is also one commit to fix URLs in four notebooks (two R and two Python) so that the eeg_eyestate_splits.csv no longer sits directly in the top of the s3 bucket.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/h2oai/h2o-3/90)
<!-- Reviewable:end -->

abal5 commented:
Can one of the admins verify this patch?

navdeep-G commented:
Can we squash these commits please?

laurendiperna commented:
now only 5 commits:
1) removed a notebook, shouldn't affect anything.
2) pubdev 3076 std_coef_plot and varimp_plot for python
3) pubdev 3176 std_coef_plot for R
4) pubdev 3121 varimp_plot for R
5) changed url to point to end of S3 bucket instead of top of bucket for a few demos, per jeff's and erin's request.

navdeep-G commented:
Screenshots of the plots would be useful

laurendiperna commented:
![screen shot 2016-08-16 at 5 03 18 pm](https://cloud.githubusercontent.com/assets/13138650/17720001/71c11a5c-63d3-11e6-800e-33b5c716b14a.png)
![screen shot 2016-08-16 at 5 03 36 pm](https://cloud.githubusercontent.com/assets/13138650/17720003/73db5c76-63d3-11e6-8e86-0066a73d2d29.png)

navdeep-G commented:
Looks good!

ledell commented:
The `std_coef_plot(self, num_of_features=None, **kwargs)` method in Python has the arguments documented in the docstring, but the `varimp_plot(self, num_of_features=None, **kwargs)` does not.  Can you add that?  You can copy-paste because they are identical.

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/h2oai/h2o-3/90#-:-KPKqkfDpTTRUZn60wab:b-48eff3)*
<!-- Sent from Reviewable.io -->

laurendiperna commented:
updated the doc string for varimp_plot() and squashed that commit into the PUBDEV-3076 commit (you can see the commit message at the bottom of that commit's list)

st-pasha commented on h2o-py/h2o/model/model_base.py:
```diff
@@ -755,6 +755,226 @@ def _plot(self, timestep, metric, **kwargs):
+            imp.find_module('six')
+            import six
+            if 'server' in list(kwargs.keys()) and kwargs['server']: six.use('Agg', warn=False)
+            from six import iteritems
```
we don't require `six` anywhere. Instead of iteritems you can use the `viewitems` function

st-pasha commented on h2o-py/h2o/model/model_base.py:
```diff
@@ -755,6 +755,226 @@ def _plot(self, timestep, metric, **kwargs):
+        """
+        # check for matplotlib. exit if absent.
+        try:
+            imp.find_module('matplotlib')
```
this is a noop (in presence of the import below)

st-pasha commented on h2o-py/h2o/model/model_base.py:
```diff
@@ -755,6 +755,226 @@ def _plot(self, timestep, metric, **kwargs):
+        try:
+            imp.find_module('matplotlib')
+            import matplotlib
+            if 'server' in list(kwargs.keys()) and kwargs['server']: matplotlib.use('Agg', warn=False)
```
The whole if condition can be written more efficiently as `if kwargs.get('server', False):`

laurendiperna commented:
@st-pasha the import six and try excepts with matplotlib were following what we had in our previous python plotting function. Maybe we should make a separate ticket to update the lines you suggested as well as the other plot function that already exists in python. how does that sound? thanks for the feed back!
